prody/proteins/interactions.py

```diff
@@ -2212,57 +2222,151 @@ def calcProteinInteractionsDCD(self, atoms, trajectory, filename=None, **kwargs)
         return HBs_nb, SBs_nb, RIB_nb, PiStack_nb, PiCat_nb, HPh_nb
 
 
-    def getInteractions(self, **kwargs):
-        """Return the list of all interactions"""
+    def getInteractions(self, filename=None, **kwargs):
+        """Return the list of all interactions.
         
-        if 'filename' in kwargs:
+        :arg filename: name of DAT file where interactions will be saved.
+        :type filename: str
+        
+        :arg selection: selection string
+        :type selection: str
+    
+        :arg selection2: selection string
+        :type selection2: str """
 
+        if len(kwargs) == 0:
+            return self._interactions_dcd
+        else:
+            sele_inter = []
+            for i in self._interactions_dcd:
+                for nr_j,j in enumerate(i):
+                    sele_inter.append(selectionByKwargs(i[nr_j], self._atoms, **kwargs))
+            self._interactions_dcd = sele_inter
+            return self._interactions_dcd
+        
+        if 'filename' != None:
```

Comment:
this should go first. Otherwise, this code is never run because both the parts above have a return

Reply:
I've fixed this